packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -478,6 +543,36 @@ class ContainerLayer extends Layer {
   Layer get lastChild => _lastChild;
   Layer _lastChild;
 
+  /// Returns whether this layer has at least one child layer.
+  bool get hasChildren => _firstChild != null;
+
+  /// Consider this layer as the root and build a scene (a tree of layers)
+  /// in the engine.
+  ui.Scene buildScene(ui.SceneBuilder builder) {
```

Comment:
kinda seems weird to have this on a layer at all. maybe we should move it to the pipeline?

Reply:
`buildScene` is used to render normal frames (i.e. the whole layer tree), but it's also used to render interior layers as if they were standalone trees, e.g. `toImage`. The pipeline seems to be designed to be 1-to-1 with the full layer tree.

The other issue is that moving it would be a breaking change.

Reply:
I forgot about toImage. Maybe add that to the docs for this method. Thanks.